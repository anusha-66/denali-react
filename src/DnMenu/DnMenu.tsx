/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import classnames from 'classnames';

export const DnMenu = ({ children, className, alignment, menuNav, ...rest }: DnMenuProps): JSX.Element => {
    return (
        <div
            className={classnames(
                'menu',
                {
                    [`is-${alignment}`]: alignment,
                    'menu--nav': menuNav,
                },
                className,
            )}
            {...rest}
        >
            {children}
        </div>
    );
};

export enum DnMenuAlignment {
    default = '',
    left = 'left',
    right = 'right',
}

export interface DnMenuProps {
    className?: string;
    children?: React.ReactNode;
    alignment?: DnMenuAlignment;
    menuNav?: string;
}

const DnMenuTrigger = ({ children, className, href, isEllipses }: DnMenuTriggerProps): JSX.Element => {
    return (
        <a href={href} className={classnames('menu-trigger', className)}>
            {isEllipses ? <i className="d-icon d-more-horizontal is-small link" /> : children}
        </a>
    );
};
DnMenu.Trigger = DnMenuTrigger;

export interface DnMenuTriggerProps {
    className?: string;
    children?: React.ReactNode;
    href?: string;
    isEllipses?: boolean;
}

const DnMenuContent = ({ children, className, ...rest }: DnMenuContentProps) => {
    return (
        <div className={classnames('menu-content', className)} {...rest}>
            {React.Children.map(children, (child) => {
                return <span className={classnames('link', className)}>{child}</span>;
            })}
        </div>
    );
};
DnMenu.Content = DnMenuContent;

export interface DnMenuContentProps {
    className?: string;
    children?: React.ReactNode;
}

DnMenu.defaultProps = {
    className: '',
    alignment: DnMenuAlignment.default,
};
