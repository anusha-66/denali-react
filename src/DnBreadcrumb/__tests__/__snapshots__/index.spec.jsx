/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnBreadcrumb } from '../../index'; // src/index.tsx
import { DnMenu } from '../../../DnMenu';

describe('DnBreadcrumb Tests', () => {
    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnBreadcrumb>
                <a href="#home">Home</a>
                <a href="#Section">Section</a>
                <a href="#SubSection">Sub Section</a>
                <span>Article</span>
            </DnBreadcrumb>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnBreadcrumb className="Foo">
                <a href="#home">Home</a>
                <a href="#Section">Section</a>
                <a href="#SubSection">Sub Section</a>
                <span>Article</span>
            </DnBreadcrumb>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with ellipses', () => {
        expect.assertions(1);

        const rendered = render(
            <DnBreadcrumb>
                <a href="#home">Home</a>
                <DnMenu>
                    <DnMenu.Trigger isEllipses />
                    <DnMenu.Content>
                        <a href="#home">Home</a>
                        <a href="#about">About</a>
                        <a href="#contact">Contact</a>
                        <a href="#help">Help</a>
                        <hr />
                        <a href="#support">Support</a>
                    </DnMenu.Content>
                </DnMenu>
                <a href="#article">Article</a>
            </DnBreadcrumb>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
