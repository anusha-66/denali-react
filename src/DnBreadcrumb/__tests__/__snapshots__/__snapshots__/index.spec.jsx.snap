// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnBreadcrumb Tests should render correctly with custom props 1`] = `
<ul
  class="breadcrumb Foo"
>
  <li>
    <span
      class="link Foo"
    >
      <a
        href="#home"
      >
        Home
      </a>
    </span>
  </li>
  <li>
    <span
      class="link Foo"
    >
      <a
        href="#Section"
      >
        Section
      </a>
    </span>
  </li>
  <li>
    <span
      class="link Foo"
    >
      <a
        href="#SubSection"
      >
        Sub Section
      </a>
    </span>
  </li>
  <li>
    <span
      class="link Foo"
    >
      <span>
        Article
      </span>
    </span>
  </li>
</ul>
`;

exports[`DnBreadcrumb Tests should render correctly with ellipses 1`] = `
<ul
  class="breadcrumb"
>
  <li>
    <span
      class="link"
    >
      <a
        href="#home"
      >
        Home
      </a>
    </span>
  </li>
  <li>
    <span
      class="link"
    >
      <div
        class="menu"
      >
        <a
          class="menu-trigger"
        >
          <i
            class="d-icon d-more-horizontal is-small link"
          />
        </a>
        <div
          class="menu-content"
        >
          <span
            class="link"
          >
            <a
              href="#home"
            >
              Home
            </a>
          </span>
          <span
            class="link"
          >
            <a
              href="#about"
            >
              About
            </a>
          </span>
          <span
            class="link"
          >
            <a
              href="#contact"
            >
              Contact
            </a>
          </span>
          <span
            class="link"
          >
            <a
              href="#help"
            >
              Help
            </a>
          </span>
          <span
            class="link"
          >
            <hr />
          </span>
          <span
            class="link"
          >
            <a
              href="#support"
            >
              Support
            </a>
          </span>
        </div>
      </div>
    </span>
  </li>
  <li>
    <span
      class="link"
    >
      <a
        href="#article"
      >
        Article
      </a>
    </span>
  </li>
</ul>
`;

exports[`DnBreadcrumb Tests should render with default props 1`] = `
<ul
  class="breadcrumb"
>
  <li>
    <span
      class="link"
    >
      <a
        href="#home"
      >
        Home
      </a>
    </span>
  </li>
  <li>
    <span
      class="link"
    >
      <a
        href="#Section"
      >
        Section
      </a>
    </span>
  </li>
  <li>
    <span
      class="link"
    >
      <a
        href="#SubSection"
      >
        Sub Section
      </a>
    </span>
  </li>
  <li>
    <span
      class="link"
    >
      <span>
        Article
      </span>
    </span>
  </li>
</ul>
`;
